rosbag2/src/rosbag2/impl/sequential_reader_impl.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "sequential_reader_impl.hpp"
+
+namespace rosbag2
+{
+
+ReaderImpl::ReaderImpl(
+  std::unique_ptr<rosbag2_storage::StorageFactoryInterface> storage_factory,
+  std::shared_ptr<SerializationFormatConverterFactoryInterface> converter_factory)
+: storage_factory_(std::move(storage_factory)),
```

Comment:
Are you sure this works? I believe you'll need a private default ctor to be called here before you overwrite the attributes. I think that as soon as you have a non-default ctor, the inline attribut init it not called anymore.